ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -1610,6 +1644,12 @@ private static Map<String, ConfigValue> buildStreamingConfig(
     final Map<String, ConfigValue> streamConfigProps = new HashMap<>();
     applyStreamsConfig(baseStreamConfig, streamConfigProps);
     applyStreamsConfig(overrides, streamConfigProps);
+
+    // Streams client metrics aren't used in Confluent deployment
+    if (streamConfigProps.get(KSQL_DEPLOYMENT_TYPE_CONFIG).value.equals(DeploymentType.confluent.name())) {
```

Comment:
I would do it the other way around just in case the value in the map is `null`
```suggestion
    if (DeploymentType.confluent.name().equals(streamConfigProps.get(KSQL_DEPLOYMENT_TYPE_CONFIG))) {
```